docker/hq-compose-runserver.yml

```diff
@@ -1,34 +1,53 @@
 version: '2.3'
 
+volumes:
+  shared_files: {}
+
+# x-anchors are ignored by Docker, are used for DRYness here
+x-commcarehq_base: &commcarehq_base
+  image: commcarehq_incl:latest
+  environment: &environment_base
+    COMMCAREHQ_BOOTSTRAP: "yes"
+    DEPENDENT_SERVICES: "couch:5984 postgres:5432 redis:6379 elasticsearch2:9200 minio:9980 kafka:9092"
+    DOCKER_HQ_OVERLAY: "${DOCKER_HQ_OVERLAY}"
+    DOCKER_HQ_OVERLAYFS_CHMOD: "${DOCKER_HQ_OVERLAYFS_CHMOD}"
+    DOCKER_HQ_OVERLAYFS_METACOPY: "${DOCKER_HQ_OVERLAYFS_METACOPY}"
+  privileged: true  # allows mount inside container
+  depends_on:
+    - postgres
+    - couch
+    - redis
+    - elasticsearch2
+    - elasticsearch5
+    - kafka
+    - zookeeper
+    - minio
+  volumes:
+    - shared_files:/sharedfiles
+
 services:
   web:
-    build:
-      context: .
-      dockerfile: "${DOCKERFILE}"
-    environment:
-      COMMCAREHQ_BOOTSTRAP: "yes"
-      DEPENDENT_SERVICES: "couch:5984 postgres:5432 redis:6379 elasticsearch2:9200 minio:9980 kafka:9092"
-      DOCKER_HQ_OVERLAY: "${DOCKER_HQ_OVERLAY}"
-      DOCKER_HQ_OVERLAYFS_CHMOD: "${DOCKER_HQ_OVERLAYFS_CHMOD}"
-      DOCKER_HQ_OVERLAYFS_METACOPY: "${DOCKER_HQ_OVERLAYFS_METACOPY}"
-    privileged: true  # allows mount inside container
-    command: [runserver]
-    depends_on:
-      - postgres
-      - couch
-      - redis
-      - elasticsearch2
-      - elasticsearch5
-      - kafka
-      - zookeeper
-      - minio
+    <<: *commcarehq_base
+    command: 'docker/start-web.sh'
     expose:
       - 8000
     ports:
       - "8000:8000"
-    volumes:
-      - ..:/mnt/commcare-hq-ro:ro
-      - ${VOLUME_PREFIX}lib:/mnt/lib
+
+  celery:
+    <<: *commcarehq_base
+    depends_on:
+      - web
```

Comment:
Curious why this depends on web? I, probably naively, think the dependency might make more sense in the opposite direction (`web` depends on `celery`, same for `pillowtop`).

Reply:
One of the HQ containers needs to run the start-up script, and the others need to wait, in order to avoid a race condition. I arbitrarily chose "web" to run the script, but it could have been one of the others. Maybe it should be "celery".

Reply:
This might or might not open a can of worms :worm: , but what if we have a separate dedicated container/service responsible for setup/startup related operations? Let's call it the `admin-service` for instance.

This way - as far as I can see
1. the web service can depend on celery/pillowtop being up and running before starting up
2. another seemingly random service (like celery) doesn't have to run the start-up script (which is a bit awkward imo)
3. (Only for when we want to make this production ready though) we can better handle access control when we start to make all this production ready - for instance, the `admin-service` can have admin related privelages, which the web service shouldn't have ideally. An example would be DB privileges: Technically the web service should not have enough privileges to alter tables/schemas. It should only be able to read/write to the DB. Another dedicated service or role should have the proper privileges to do migrations etc. Having this service might also prove to make migrations easier down the line when new changes are rolled out, since the service doesn't have to run in a broken state until all migrations have run. Anycase, a chat for another day.

Point 3 is deffs overkill for now, but thoughts on the first two?